Ash.FrontEnd/src/js/global/components/Fido.jsx

```diff
@@ -0,0 +1,293 @@
+import React from 'react';
+import memoizeOne from 'memoize-one';
+
+/* EXAMPLE
+fetchItems = {
+  name: {
+    // ...configObject
+  },
+  nextName,
+  ...
+}
+*/
+
+/**
+ * Fetch configuration object default object shape
+ * @prop {string} path - fetch url
+ * @prop {object} query - object containing query string key-value pairs
+ * @prop {object} payload - json payload for request
+ * @prop {object} options - other options [optional]
+ * @prop {string} options.method - HTTP method type (case-insensitive) [optional]
+ * @prop {boolean} options.callOnMount - flag for whether the fetch should be
+ *    performed on mount [optional]
+ * @prop {boolean} options.preservePreviousData - flag for whether the fetch should
+ *    preserve the previously returned data set when a new call is initiated [optional]
+ */
+const defaultFetchConfig = {
+  path: '',
+  query: null,
+  payload: null,
+  options: {
+    method: 'get',
+    callOnMount: true,
+    preservePreviousData: true,
+  },
+};
+
+/**
+ * Fetch state object default object shape
+ * @prop {boolean} inFlight - is the request awaiting a reply?
+ * @prop {boolean} success - did we receive a good reply from the server?
+ * @prop {boolean} fail - did the request fail or did the server return a bad response?
+ * @prop {object} data - response payload of the successful request
+ * @prop {object} error - js error object
+ * @prop {Function} call - function to initiate a fetch
+ */
+const initialFetchState = {
+  inFlight: false,
+  success: false,
+  fail: false,
+  data: null,
+  error: null,
+  call: null,
+};
+
+/**
+ * Generate fetch state objects
+ * @param {object} fetchItems - Object containing named config objects
+ * @param {object} callFunc - "call" function
```

Comment:
What does "call" in quotes like that mean exactly? It's late and I'm probably just tired but not sure what you're getting at

Reply:
It's for initiating the fetch.  I'll make the comment more explicit.

Reply:
Also don't see any changes made for this

Reply:
I've attempted to improve the comment.